Shortcut to open policy editor from process config panel

nkylstad opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
- Added the option to set a default tab when opening the settings modal
- Added an `AppDevelopmentContext` to app-development application.
  - The context currently only contains information about the settings modal:
    - If it is open
    - Which tab should be the default
- Pass a function that utilizes the settings modal state from the context to the process editor
- The process editor has a button that updates the state in the global (appDevelopment) context and opens the settingsModal with the policy editor as default
  - This is shown in a new accordion for policy rules in the config panel (see screenshots below)

### Reasoning for this solution
- It is likely we will need a "global" context for app-development application in the future. F.ex. for storing global validation state for the entire app. 
- The changes needed for this issue are quite minimal with this approach
- No need for url rewrite complexity when dealing with a modal

Happy to discuss the proposed solution. 
Would also like some feedback on texts and look, as this is currently very basic. 

<img width="1508" alt="Screenshot 2024-05-13 at 20 51 48" src="https://github.com/Altinn/altinn-studio/assets/1636323/dc7f5e9c-15cc-4a79-a966-b4a8274fe511">

<img width="423" alt="Screenshot 2024-05-13 at 20 51 55" src="https://github.com/Altinn/altinn-studio/assets/1636323/12b7a274-4490-4d8c-9aa7-b67932451d0a">


## Related Issue(s)

- #12744 

## Verification

- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)

## Documentation

- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

framitdavid commented:
You have done some great reasoning by yourself and I do agree! 🚀 

At first, I thought that URL-based state would be nice. Just to give the service owner the availability to share the "page" they are looking at with their colleagues (f.ex when they are asking for help). But, I do agree that URL complexity might not be worth it. Let's keep it simple, just like your implementation within this PR.😊


**Global Context**
I do agree that we need a global app-context for global validations and controlling states such as the settings modal. I'm little scared that this context can and will be quite large and complex over time. I believe that this context will grow in states, methods and side-effects, and then it becomes hard to debug and maintain. 

Due to my concerns above, I want to challenge the implementation of the app-context to make it more modular. With modular I mean one context for each domain/concern to solve. Hehe, and then we hit the Provider hell... 😆However, I like [this short blogpost which gives us a great example of combined contexts.](https://javascript.plainenglish.io/how-to-combine-context-providers-for-cleaner-react-code-9ed24f20225e)

By following a pattern like this, we get just one single app-context-provider at root and we can isolate each module/context with the responsibility of their domain. Makes it easier to read each of the contexts. Note: Feel free to disagree up-on this suggestion!😊

nkylstad commented:
> Due to my concerns above, I want to challenge the implementation of the app-context to make it more modular. With modular I mean one context for each domain/concern to solve. Hehe, and then we hit the Provider hell... 😆However, I like [this short blogpost which gives us a great example of combined contexts.](https://javascript.plainenglish.io/how-to-combine-context-providers-for-cleaner-react-code-9ed24f20225e)
> 
> By following a pattern like this, we get just one single app-context-provider at root and we can isolate each module/context with the responsibility of their domain. Makes it easier to read each of the contexts. Note: Feel free to disagree up-on this suggestion!😊

I completely agree, and this is something I thought might become an issue as well! Thanks for the input, this looks like a great approach to split the context into more domain-specific chunks. I think it makes it easier to use as well, as it will be obvious which context to use depending on the need of each particular case/component.
I have implemented the suggested pattern now!

framitdavid commented on frontend/app-development/contexts/AppDevelopmentContext.tsx:
```diff
@@ -1,44 +1,12 @@
+import { SettingsModalContextProvider } from './SettingsModalContext';
+import { combineComponents } from 'app-development/utils/context/combineComponents';
+
+/**
```
This is a great usage of comments! Important information. 👏

framitdavid reviewed (commented):
Solid work! 🚀

nkylstad commented:
@Annikenkbrathen I have looked at the existing sketches for process editor, and the closest match for this functionality (open settings modal, with policy editor tab selected, from process editor) looks like the "Legg til ny datamodell" button. I have updated my implementation to match that. See video below.

https://github.com/Altinn/altinn-studio/assets/1636323/4a62b39f-4ba7-49b7-91e7-3b8ca1732a62

An alternative might be to have a more explicit button, like the design for "Gå til lage"-button for CustomReceipt:
![Screenshot 2024-05-14 at 11 53 06](https://github.com/Altinn/altinn-studio/assets/1636323/2c521ebe-69af-4519-af22-14b5a2519ede)

Which would be something like: 
![Screenshot 2024-05-14 at 12 01 37](https://github.com/Altinn/altinn-studio/assets/1636323/bd8b090f-4d2b-4133-a05d-5f4d1bb1cd3a)

Thoughts?

Annikenkbrathen commented:
This is very nice!😊I like the sectioning that this exists in its own accordion.
A question: Have we considered whether we can validate if this task is already covered by the existing policy rules? If we can determine if it has been included, we can formulate the message accordingly.

The component to open the settingsmodal:
Yes, I have created a component that can be used for navigating across pages. The question is whether we should use the same component to open a modal as we use to navigate to a new page? I think we can use the same one for now.
In the sketches, I'm designing a white settings panel, so this component now has a gray background in Figma. In Studio, we still use a gray background for the settings panel, so William and I decided to have a white background for this component to ensure the correct contrast.

![Skjermbilde 2024-05-14 kl  13 40 29](https://github.com/Altinn/altinn-studio/assets/100292724/d2353f0f-a203-428c-9de9-1ef670e54e0c)
https://www.figma.com/design/VAvGOqkMhKM8HL8h4xBeDH7b/Re-design-Altinn-Studio?m=dev&node-id=11459%3A19563

@Ildest might have some comments on the text here.
My suggestion for the alert message is: 
"Tilgangsreglene som lages for appen må dekke over alle oppgavene. Sjekk om du har en regel som dekker denne oppgaven i app-innstillingene. Om den ikke er dekket, kan du enten lage en ny regel eller inkludere denne oppgaven i en allerede eksisterende regel."

And my suggestion for the button and heading is in the sketch above.
Also, We are changing the tab name from "tilgangsregler" to "tilganger". Because it's easier to relate to, and do contain more than the police rules config, like "sikkerhetnivå". Right @Ildest?

A digression:
In the policy editor sketches that we are working on, we are calling it "steg" for now. Should we call it "oppgave" instead?
![Skjermbilde 2024-05-14 kl  13 45 48](https://github.com/Altinn/altinn-studio/assets/100292724/3d9cf9ff-41cd-48f5-b5d7-aa8cf628ba8d)

nkylstad commented:
> This is very nice!😊I like the sectioning that this exists in its own accordion. A question: Have we considered whether we can validate if this task is already covered by the existing policy rules? If we can determine if it has been included, we can formulate the message accordingly.

Yes, our suggestion here was to implement this step-wise:
- First completely unaware, with a shortcut to open the policy editor. 
- Then add an overview of which rules apply for the given task. This is probably quite complicated, as some rules apply for the entire app, some for just the selected task, and some might apply for a set of tasks. If we do this, then we could differentiate between the different kinds of rules and show which ones apply to the selected task.

So for the first step, the message would have to be generic.

> The component to open the settingsmodal: Yes, I have created a component that can be used for navigating across pages. The question is whether we should use the same component to open a modal as we use to navigate to a new page? I think we can use the same one for now. In the sketches, I'm designing a white settings panel, so this component now has a gray background in Figma. In Studio, we still use a gray background for the settings panel, so William and I decided to have a white background for this component to ensure the correct contrast.

Nice! The background of the config panel for the process editor is white though 🤔 This is from dev.altinn.studio:
![Screenshot 2024-05-14 at 14 27 48](https://github.com/Altinn/altinn-studio/assets/1636323/f09b49c9-ddea-42d4-9e08-8ea0325cd8ad)

> @Ildest might have some comments on the text here. My suggestion for the alert message is: "Tilgangsreglene som lages for appen må dekke over alle oppgavene. Sjekk om du har en regel som dekker denne oppgaven i app-innstillingene. Om den ikke er dekket, kan du enten lage en ny regel eller inkludere denne oppgaven i en allerede eksisterende regel."

👍 

> And my suggestion for the button and heading is in the sketch above. Also, We are changing the tab name from "tilgangsregler" to "tilganger". Because it's easier to relate to, and do contain more than the police rules config, like "sikkerhetnivå". Right @Ildest?

👍 

> A digression: In the policy editor sketches that we are working on, we are calling it "steg" for now. Should we call it "oppgave" instead? ![Skjermbilde 2024-05-14 kl 13 45 48](https://private-user-images.githubusercontent.com/100292724/330410758-3d9cf9ff-41cd-48f5-b5d7-aa8cf628ba8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2ODgxOTksIm5iZiI6MTcxNTY4Nzg5OSwicGF0aCI6Ii8xMDAyOTI3MjQvMzMwNDEwNzU4LTNkOWNmOWZmLTQxY2QtNDhmNS1iNWQ3LWFhOGNmNjI4YmE4ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxMTU4MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMmQ2ZGU5YmRmYTAwNWM4YzM1Y2IxOWEzZGQ2ZjgxNWJiYWQyZDljN2Q2MGFmN2MxODM2MWY3NWYwMzEzZjYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kNieYzcZLEyhOCDvvxbohf2nH5F52Z4_UfdmVShMc8k)

Yes probably, because that is the term used in process modelling 🤔

Annikenkbrathen commented:
> Yes, our suggestion here was to implement this step-wise:
> 
> * First completely unaware, with a shortcut to open the policy editor.
> * Then add an overview of which rules apply for the given task. This is probably quite complicated, as some rules apply for the entire app, some for just the selected task, and some might apply for a set of tasks. If we do this, then we could differentiate between the different kinds of rules and show which ones apply to the selected task.
> 
> So for the first step, the message would have to be generic.
> 

Sounds like a good idea!

> > The component to open the settingsmodal: Yes, I have created a component that can be used for navigating across pages. The question is whether we should use the same component to open a modal as we use to navigate to a new page? I think we can use the same one for now. In the sketches, I'm designing a white settings panel, so this component now has a gray background in Figma. In Studio, we still use a gray background for the settings panel, so William and I decided to have a white background for this component to ensure the correct contrast.
> 
> Nice! The background of the config panel for the process editor is white though 🤔 This is from dev.altinn.studio: ![Screenshot 2024-05-14 at 14 27 48](https://private-user-images.githubusercontent.com/1636323/330425132-f09b49c9-ddea-42d4-9e08-8ea0325cd8ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2OTI3NTcsIm5iZiI6MTcxNTY5MjQ1NywicGF0aCI6Ii8xNjM2MzIzLzMzMDQyNTEzMi1mMDliNDljOS1kZGVhLTQyZDQtOWUwOC04ZWEwMzI1Y2Q4YWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTRUMTMxNDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGU1MDdkNDcwNmY0ZTY5MjVlNGQ4YjQ5NzMxYTQ5MzNjN2FkZWM5NmVlNzY2OGEzYWUwMGViZjgwNjkzNDNlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.bl3WIVnEqh6XJ_MDGCpSrWaFSzE47F1bzJbXwauosoU)

Omg! then I have been confused. So he made it gray on purpose?!! I thought it was from the old design or something. I think we should continue  with white background if possible and have the component in gray as you shown Nina.

<img width="146" alt="Skjermbilde 2024-05-14 kl  15 15 34" src="https://github.com/Altinn/altinn-studio/assets/100292724/5130c738-63e5-472f-a607-b8674f948058">
Then we need to fix the colors here. So sorry that I didn't realized that! 😭

> > A digression: In the policy editor sketches that we are working on, we are calling it "steg" for now. Should we call it "oppgave" instead? ![Skjermbilde 2024-05-14 kl 13 45 48](https://private-user-images.githubusercontent.com/100292724/330410758-3d9cf9ff-41cd-48f5-b5d7-aa8cf628ba8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2ODgxOTksIm5iZiI6MTcxNTY4Nzg5OSwicGF0aCI6Ii8xMDAyOTI3MjQvMzMwNDEwNzU4LTNkOWNmOWZmLTQxY2QtNDhmNS1iNWQ3LWFhOGNmNjI4YmE4ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxMTU4MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMmQ2ZGU5YmRmYTAwNWM4YzM1Y2IxOWEzZGQ2ZjgxNWJiYWQyZDljN2Q2MGFmN2MxODM2MWY3NWYwMzEzZjYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kNieYzcZLEyhOCDvvxbohf2nH5F52Z4_UfdmVShMc8k)

> 
> Yes probably, because that is the term used in process modelling 🤔
Great

nkylstad commented:
> Omg! then I have been confused. So he made it gray on purpose?!! I thought it was from the old design or something. I think we should continue with white background if possible and have the component in gray as you shown Nina.
> 
> <img alt="Skjermbilde 2024-05-14 kl 15 15 34" width="146" src="https://private-user-images.githubusercontent.com/100292724/330445192-5130c738-63e5-472f-a607-b8674f948058.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2OTUxMDgsIm5iZiI6MTcxNTY5NDgwOCwicGF0aCI6Ii8xMDAyOTI3MjQvMzMwNDQ1MTkyLTUxMzBjNzM4LTYzZTUtNDcyZi1hNjA3LWI4Njc0Zjk0ODA1OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxMzUzMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MWVmZjU0NWY3NDMyOGZmZjg1MDBkOGQxNDUzYzkxYWRmNWE0N2RmOGFkYjcwMWNhMjE2NWEzZjdjMzY0YTg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.CO80729UkfBDZefRYY645OBCJ0uGDKESxJV_nbaIhwI"> Then we need to fix the colors here. So sorry that I didn't realized that! 😭

Not entirely sure if the white background is a recent update or not, but it's in all environments at least 🤷‍♀️ 
Entirely possible that it is fairly recent and just a matter of an older branch 😄 

In any case I will use the grey background then, and the texts you have suggested. Will post a video when it's done, and then I think we are good to go for now?

Ildest commented:
I will have a look at the texts in this issue today @nkylstad :-).

Ildest commented:
> "Tilgangsreglene som lages for appen må dekke over alle oppgavene. Sjekk om du har en regel som dekker denne oppgaven i app-innstillingene. Om den ikke er dekket, kan du enten lage en ny regel eller inkludere denne oppgaven i en allerede eksisterende regel."

I hope I get this answer quoted o the correct thread :-D. Departing from Ninas suggestion and then incorporating "steg" instead of "oppgave, I will suggest this: "Tilgangsreglene du lager må dekke alle stegene. Gå till innstillingene for appen og sjekk om du har en regel som dekker dette steget. Hvis det ikke er dekket, lager du enten en ny regel eller inkluderer dette steget i en regel som finnes fra før."

Annikenkbrathen commented:
> "Tilgangsreglene som lages for appen må dekke over alle oppgavene. Sjekk om du har en regel som dekker denne oppgaven i app-innstillingene. Om den ikke er dekket, kan du enten lage en ny regel eller inkludere denne oppgaven i en allerede eksisterende regel."
> I hope I get this answer quoted o the correct thread :-D. Departing from Ninas suggestion and then incorporating "steg" instead of "oppgave, I will suggest this: "Tilgangsreglene du lager må dekke alle stegene. Gå till innstillingene for appen og sjekk om du har en regel som dekker dette steget. Hvis det ikke er dekket, lager du enten en ny regel eller inkluderer dette steget i en regel som finnes fra før."

We meant the opposite, Gørild. We have to change from "Steg" to "Oppgave" in our text since we refer to the task as "oppgave" on the process page.

Therefore, the text will end up being this after your corrections @Ildest 
"Tilgangsreglene du lager må dekke alle oppgavene. Gå till innstillingene for appen og sjekk om du har en regel som dekker denne oppgaven. Hvis det ikke er dekket, lager du enten en ny regel eller inkluderer denne oppgaven i en regel som finnes fra før."

Annikenkbrathen commented:
New text suggestion after discussing with @Ildest 
<img width="749" alt="Skjermbilde 2024-05-15 kl  09 26 13" src="https://github.com/Altinn/altinn-studio/assets/100292724/f0b5f779-de14-4727-9e61-66c0735b2e67">

And its also important that we change the name in the tab from Tilgangsregler to Tilganger, so the titles do match.
<img width="729" alt="Skjermbilde 2024-05-15 kl  10 57 08" src="https://github.com/Altinn/altinn-studio/assets/100292724/222cc03f-df5f-4649-bc5f-c526a13683ea">

nkylstad commented:
@Annikenkbrathen Updated design implementation and texts now, this is what it looks like:
![Screenshot 2024-05-15 at 12 16 07](https://github.com/Altinn/altinn-studio/assets/1636323/e9a93401-147d-4443-8ade-f8ad394cd521)

Also updated the tab heading to match in settings modal: 

![Screenshot 2024-05-15 at 12 15 59](https://github.com/Altinn/altinn-studio/assets/1636323/5813df3b-551d-4561-bdc3-9a1ea67ac75f)

Annikenkbrathen commented:
LGTM 🤩

WilliamThorenfeldt reviewed (commented):
Great job! 😄 
Added a few comments

nkylstad commented:
> Great job! 😄 Added a few comments

@WilliamThorenfeldt I have updated where you have commented now 😊 
I think it's my editor that sets these paths, we might consider setting up a linting rule if this is something we want to enforce, so we don't have to check for it manually. I found at least 1 case in the files I changed now that was not added by me 😅

framitdavid commented on frontend/app-development/contexts/SettingsModalContext.test.tsx:
```diff
@@ -0,0 +1,47 @@
+import { SettingsModalContextProvider, useSettingsModalContext } from './SettingsModalContext';
+
+describe('SettingsModalContext', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
```
As far as I understand, it's unnecessary to have clearAllMocks when mocks are defined within each test and its respective it/test block. 🤔

nkylstad replied:
Yep, I agree. I have removed this.

WilliamThorenfeldt commented:
> > Great job! 😄 Added a few comments
> 
> @WilliamThorenfeldt I have updated where you have commented now 😊 I think it's my editor that sets these paths, we might consider setting up a linting rule if this is something we want to enforce, so we don't have to check for it manually. I found at least 1 case in the files I changed now that was not added by me 😅

I agree that we should try to implement some linting rules to fix that 😄 
There is probably some areas in our code that still is not updated in regards of this yes.

WilliamThorenfeldt commented on frontend/app-development/test/testUtils.tsx:
```diff
@@ -12,6 +12,7 @@ import { ServicesContextProvider } from 'app-shared/contexts/ServicesContext';
 import { queryClientConfigMock } from 'app-shared/mocks/queryClientMock';
 import type { QueryClient } from '@tanstack/react-query';
 import { queriesMock } from 'app-shared/mocks/queriesMock';
+import { AppDevelopmentContextProvider } from 'app-development/contexts/AppDevelopmentContext';
```
Should change this import 😄

WilliamThorenfeldt commented on frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditPolicy/EditPolicy.tsx:
```diff
@@ -0,0 +1,36 @@
+import { StudioButton } from '@studio/components';
+import { useBpmnApiContext } from '../../../../contexts/BpmnApiContext';
+import { useTranslation } from 'react-i18next';
+
```
```suggestion
```

WilliamThorenfeldt reviewed (commented):
Still some places where there is absolute imports instead of relative imports 😄

nkylstad commented:
> Still some places where there is absolute imports instead of relative imports 😄

Lol I had somehow not pushed the changes up from my machine 🙈 
Pushed them now!

Ildest commented:
> This is very nice!😊I like the sectioning that this exists in its own accordion. A question: Have we considered whether we can validate if this task is already covered by the existing policy rules? If we can determine if it has been included, we can formulate the message accordingly.
> 
> The component to open the settingsmodal: Yes, I have created a component that can be used for navigating across pages. The question is whether we should use the same component to open a modal as we use to navigate to a new page? I think we can use the same one for now. In the sketches, I'm designing a white settings panel, so this component now has a gray background in Figma. In Studio, we still use a gray background for the settings panel, so William and I decided to have a white background for this component to ensure the correct contrast.
> 
> ![Skjermbilde 2024-05-14 kl 13 40 29](https://private-user-images.githubusercontent.com/100292724/330409220-d2353f0f-a203-428c-9de9-1ef670e54e0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3NTQwMzksIm5iZiI6MTcxNTc1MzczOSwicGF0aCI6Ii8xMDAyOTI3MjQvMzMwNDA5MjIwLWQyMzUzZjBmLWEyMDMtNDI4Yy05ZGU5LTFlZjY3MGU1NGUwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNVQwNjE1MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNDg0ZTQ0ZTQyYzVhZDMyOTlkYTk2ZDM1NGZjMjhiYTY5ZDg0YjQ5OTQ4NGEwZjA3NzA3MzZhM2NlY2ZjYTkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.c9t7z81uYDBlgRS-hI82AOrmbeNvX6ZljUrC4d_ECxU) https://www.figma.com/design/VAvGOqkMhKM8HL8h4xBeDH7b/Re-design-Altinn-Studio?m=dev&node-id=11459%3A19563
> 
> @Ildest might have some comments on the text here. My suggestion for the alert message is: "Tilgangsreglene som lages for appen må dekke over alle oppgavene. Sjekk om du har en regel som dekker denne oppgaven i app-innstillingene. Om den ikke er dekket, kan du enten lage en ny regel eller inkludere denne oppgaven i en allerede eksisterende regel."
> 
> And my suggestion for the button and heading is in the sketch above. Also, We are changing the tab name from "tilgangsregler" to "tilganger". Because it's easier to relate to, and do contain more than the police rules config, like "sikkerhetnivå". Right @Ildest?
> 
> A digression: In the policy editor sketches that we are working on, we are calling it "steg" for now. Should we call it "oppgave" instead? ![Skjermbilde 2024-05-14 kl 13 45 48](https://private-user-images.githubusercontent.com/100292724/330410758-3d9cf9ff-41cd-48f5-b5d7-aa8cf628ba8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3NTQwMzksIm5iZiI6MTcxNTc1MzczOSwicGF0aCI6Ii8xMDAyOTI3MjQvMzMwNDEwNzU4LTNkOWNmOWZmLTQxY2QtNDhmNS1iNWQ3LWFhOGNmNjI4YmE4ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNVQwNjE1MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNTM2YWI2YTM4NDk0NTc0YTRhYTYzMjM3NjFiMWVhYzA4MjdmMWY4YmRhZTgwZDQ2NGU4YzBlNDY1MTY1ZGYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gWkvL8-ykiM8C_uHlvF1YCCrv0VO27qQqWTq0mT_5sg)

@Annikenkbrathen For this:
![image](https://github.com/Altinn/altinn-studio/assets/111488506/900020af-6f80-4b91-a059-abd6db6917a7)


I would suggest:
Du må ha tilgangsregler som dekker alle oppgaver. Gå til Tilganger for å sjekke om du har en regel som dekker denne oppgaven.  Hvis du ikke har en regel for oppgaven, kan du enten lage en ny regel eller inkludere denne oppgaven i en regel som allerede finnes.

Ildest commented:
@nkylstad hei! Denne saken ble dessverre lukket uten at tekstforslaget mitt kom med - jeg så ikke at kommentaren min ikke var blitt postet på saken før nå. Har du mulighet til å gjenåpne, fikse eventuelt?

nkylstad commented:
> @nkylstad hei! Denne saken ble dessverre lukket uten at tekstforslaget mitt kom med - jeg så ikke at kommentaren min ikke var blitt postet på saken før nå. Har du mulighet til å gjenåpne, fikse eventuelt?

@Ildest Har lagt inn en PR med den oppdaterte teksten nå! https://github.com/Altinn/altinn-studio/pull/12860
